app/services/waste_exemptions_engine/determine_easting_and_northing_service.rb

```diff
@@ -34,27 +34,34 @@ def easting_and_northing_from_grid_reference(grid_reference)
     def easting_and_northing_from_postcode(postcode)
       return if postcode.blank?
 
-      results = AddressFinderService.new(postcode).search_by_postcode
+      response = AddressLookupService.run(postcode)
 
-      if results.is_a?(Symbol)
-        default_do_not_fetch_again_coordinates
+      if response.successful?
+        @result[:easting] = response.results.first["x"].to_f
+        @result[:northing] = response.results.first["y"].to_f
 
         return
       end
 
-      no_result_from_postcode_lookup(postcode) && return if results.empty?
+      no_match_from_postcode_lookup(postcode) && return if response.error.is_a?(DefraRuby::Address::NoMatchError)
```

Comment:
I feel like this could all be cleaned up a bit to be easier to read, as at the moment we essentially have an `if/elsif/else` block but it's all structured differently. Maybe something like:

```
if response.successful?
  @result[:easting] = response.results.first["x"].to_f
  @result[:northing] = response.results.first["y"].to_f
elsif response.error.is_a?(DefraRuby::Address::NoMatchError)
  no_match_from_postcode_lookup(postcode)
else
  error_from_postcode_lookup(postcode, response.error)
end
```

Or maybe even some further refactoring, like:

```
if response.successful?
  set_result_coordinates_from_postcode_lookup(response)
elsif response.error.is_a?(DefraRuby::Address::NoMatchError)
  no_match_from_postcode_lookup(postcode)
else
  error_from_postcode_lookup(postcode, response.error)
end
```

What do you think?

Reply:
I think I was trying to be too clever and use ruby idioms 😁 ! I think that makes a lot of sense.